zebra-network/src/peer_set/set/tests/prop.rs

```diff
@@ -90,17 +96,85 @@ proptest! {
             Ok::<_, TestCaseError>(())
         })?;
     }
+
+    /// Test if requests are broadcasted to the right number of peers.
+    #[test]
+    fn broadcast_to_peers(
+        peer_versions in any::<PeerVersions>(),
+    ) {
+        // Get a dummy block hash to help us construct a valid request to be broadcasted
+        let block: block::Block = zebra_test::vectors::BLOCK_MAINNET_10_BYTES
+            .zcash_deserialize_into()
+            .unwrap();
+        let block_hash = block::Hash::from(&block);
+
+        // Get the current valid peer version
+        let current_mainnet_version = Version::min_specified_for_upgrade(
+            Network::Mainnet,
+            NetworkUpgrade::Canopy
+        );
+
+        // Start the runtime
+        let runtime = zebra_test::init_async();
+        let _guard = runtime.enter();
+
+        // Get peers and handles
+        let (discovered_peers, mut handles) = peer_versions.mock_peer_discovery();
+        let (minimum_peer_version, _best_tip_height) =
+            MinimumPeerVersion::with_mock_chain_tip(Network::Mainnet);
+
+        // Build a peerset
+        runtime.block_on(async move {
+            let (mut peer_set, _peer_set_guard) = PeerSetBuilder::new()
+                .with_discover(discovered_peers)
+                .with_minimum_peer_version(minimum_peer_version.clone())
+                .build();
+
+            // Get the total number of active peers
+            let total_number_of_active_peers = check_if_only_up_to_date_peers_are_live(
+                &mut peer_set,
+                &mut handles,
+                current_mainnet_version,
+            )?;
+
+            // Exit the test early if the total number of active peers is zero
+            if total_number_of_active_peers == 0 {
+                return Ok::<_, TestCaseError>(());
+            }
+
+            // Get the total number of peers to broadcast
+            let number_of_peers_to_broadcast = peer_set.number_of_peers_to_broadcast();
+
+            // Send a request to all peers
+            let _ = peer_set.route_broadcast(Request::AdvertiseBlock(block_hash));
+
+            // Check how many peers received the request
+            let mut received = 0;
+            for mut h in handles {
+                if let ReceiveRequestAttempt::Request(client_request) = h.try_to_receive_outbound_client_request() {
+                    assert_eq!(client_request.request, Request::AdvertiseBlock(block_hash));
```

Comment:
Can we use `prop_assert_eq!` in this test?

It gives better debugging output.

Reply:
https://github.com/ZcashFoundation/zebra/pull/3284/commits/fb8483287627b0afff0103488e5105d1a5d20f19